Core/FunctionTree/Functions.hpp

```diff
@@ -29,9 +29,9 @@ namespace FunctionTree {
 ///
 class Strategy {
 public:
-  Strategy(ParType in, std::string op = "") : checkType(in), Op(op){};
```

Comment:
Just a naming thing: not entirely clear to me what `op` is?

Reply:
Hmm true, I could have also improved that. If there is some more stuff change, then I can do that. Otherwise I feel a bit lazy to update the PR just for this

Reply:
Nah is fine, just took me a bit more effort to understand the code

Reply:
Ok is fixed